### ISSUE TITLE
Username registration does not display immediately

### ISSUE THREAD

**Issue by holmesworcester** [ZbayApp/zbay#377]
I'm not exactly sure about the steps to reproduce for this, but I've seen it happen several times:

1. A new user registers an account
2. The user sends a message
3. The user's first new message appears to other online as being from an anonXXXXX username. 
4. (I think) restarting or waiting will show the message to appear as from the user. 

The strange thing is that the zbay channel knows there's a new user and that they've registered, but Zbay hasn't incorporated their username registration yet.

**Comment by holmesworcester** [comment]
@NorbertBodziony thoughts on fixes for this? this is confusing quite a few users so it'd be good to fix soon!

**Comment by NorbertBodziony** [comment]
Maybe its good idea to show loader? We can set nickname immediately but it will not reflect network state.

**Comment by NorbertBodziony** [comment]
Also when we receive message we check if there is person that matches signature provided with message but when user sends message and then register username his username will not appear even if registration confirms on network. One way to solve this is to refresh messages or update messages that match signature of new user. Of course if user restarts application his old messages should appear with nickname.

**Comment by holmesworcester** [comment]
Yes I think once we receive a user registration message we should check to
see if we have any existing messages with a matching signature.

**Comment by NorbertBodziony** [comment]
What about user should change nickname immediately ?
